ray-operator/controllers/ray/rayservice_controller_unit_test.go

```diff
@@ -93,102 +93,241 @@ func TestGenerateHashWithoutReplicasAndWorkersToDelete(t *testing.T) {
+
+	tests := []struct {
+		rayService        *rayv1.RayService
+		activeRayCluster  *rayv1.RayCluster
+		pendingRayCluster *rayv1.RayCluster
+		name              string
+		expectedAction    ClusterAction
+	}{
+		{
+			name: "Has pending cluster name and cluster spec is the same",
+			rayService: &rayv1.RayService{
+				Spec: rayv1.RayServiceSpec{
+					RayClusterSpec: rayClusterBase.Spec,
+				},
+				Status: rayServiceStatusWithPendingCluster,
+			},
+			activeRayCluster:  nil,
+			pendingRayCluster: rayClusterBase,
+			expectedAction:    DoNothing,
+		},
+		{
+			name: "Has pending cluster name and cluster spec has different Ray version",
+			rayService: &rayv1.RayService{
+				Spec: rayv1.RayServiceSpec{
+					RayClusterSpec: rayClusterDifferentRayVersion.Spec,
+				},
+				Status: rayServiceStatusWithPendingCluster,
+			},
+			activeRayCluster:  nil,
+			pendingRayCluster: rayClusterBase,
+			expectedAction:    CreatePendingCluster,
+		},
+		{
+			name: "Has pending cluster name and cluster spec has different replicas and workers to delete",
+			rayService: &rayv1.RayService{
+				Spec: rayv1.RayServiceSpec{
+					RayClusterSpec: rayClusterDifferentReplicasAndWorkersToDelete.Spec,
+				},
+				Status: rayServiceStatusWithPendingCluster,
+			},
+			activeRayCluster:  nil,
+			pendingRayCluster: rayClusterBase,
+			expectedAction:    DoNothing,
+		},
+		{
+			name: "Has pending cluster name and cluster spec has different worker group name",
+			rayService: &rayv1.RayService{
+				Spec: rayv1.RayServiceSpec{
+					RayClusterSpec: rayClusterDifferentWorkerGroup.Spec,
+				},
+				Status: rayServiceStatusWithPendingCluster,
+			},
+			activeRayCluster:  nil,
+			pendingRayCluster: rayClusterBase,
+			expectedAction:    CreatePendingCluster,
+		},
+		{
+			name: "Has pending cluster name and cluster spec has additional worker group",
+			rayService: &rayv1.RayService{
+				Spec: rayv1.RayServiceSpec{
+					RayClusterSpec: rayClusterAdditionalWorkerGroup.Spec,
+				},
+				Status: rayServiceStatusWithPendingCluster,
+			},
+			activeRayCluster:  nil,
+			pendingRayCluster: rayClusterBase,
+			expectedAction:    UpdatePendingCluster,
+		},
+		{
+			name: "Has pending cluster name and cluster spec has no worker group",
+			rayService: &rayv1.RayService{
+				Spec: rayv1.RayServiceSpec{
+					RayClusterSpec: rayClusterWorkerGroupRemoved.Spec,
+				},
+				Status: rayServiceStatusWithPendingCluster,
+			},
+			activeRayCluster:  nil,
+			pendingRayCluster: rayClusterBase,
+			expectedAction:    CreatePendingCluster,
+		},
+		{
+			name:              "No pending cluster name and no active cluster",
+			rayService:        &rayv1.RayService{},
+			activeRayCluster:  nil,
+			pendingRayCluster: nil,
+			expectedAction:    GeneratePendingClusterName,
+		},
+		{
+			name:              "No pending cluster name and active cluster has different kubeRayVersion",
```

Comment:
nit
```suggestion
			name:              "No pending cluster name and active cluster has different KubeRayVersion",
```

Reply:
Fixed in https://github.com/ray-project/kuberay/pull/2716/commits/126b6ad72174750da65742d32c99392597b91687